Example/Example/UI/Weather/WeatherViewController.swift

```diff
@@ -8,94 +8,109 @@
 
 import UIKit
 
-protocol WeatherModel {
-    func fetchWeather(at area: String, date: Date, completion: @escaping (Result<Response, WeatherError>) -> Void)
-}
-
-protocol DisasterModel {
-    func fetchDisaster(completion: ((String) -> Void)?)
-}
-
-class WeatherViewController: UIViewController {
+final class WeatherViewController: UIViewController {
+    
+    private let weatherModel: WeatherModel
+    private let disasterModel: DisasterModel
+    private let weatherHandler: WeatherHandler
+    private let mainQueueScheduler: QueueScheduler
+    private let globalQueueScheduler: QueueScheduler
     
-    var weatherModel: WeatherModel!
-    var disasterModel: DisasterModel!
     @IBOutlet weak var weatherImageView: UIImageView!
     @IBOutlet weak var minTempLabel: UILabel!
     @IBOutlet weak var maxTempLabel: UILabel!
-    @IBOutlet weak var disasterLabel: UILabel!
-    @IBOutlet weak var activityIndicator: UIActivityIndicatorView!
+    @IBOutlet weak private var disasterLabel: UILabel!
+    @IBOutlet weak private var activityIndicator: UIActivityIndicatorView!
+    @IBOutlet weak private var reloadButton: UIButton!
+    @IBOutlet weak private var closeButton: UIButton!
     
     override func viewDidLoad() {
         super.viewDidLoad()
-        
-        NotificationCenter.default.addObserver(forName: UIApplication.didBecomeActiveNotification, object: nil, queue: nil) { [unowned self] notification in
-            self.loadWeather(notification.object)
-        }
+        NotificationCenter.default.addObserver(self, selector: #selector(reload), name: UIApplication.didBecomeActiveNotification, object: nil)
+    }
+    
+    init?(coder: NSCoder, weatherModel: WeatherModel, disasterModel: DisasterModel = DisasterModelImpl(), weatherHandler: WeatherHandler = .live, mainQueueScheduler: QueueScheduler = .main, globalQueueScheduler: QueueScheduler = .global) {
```

Comment:
[NITS] 長くなりすぎる場合は適宜改行など
```suggestion
    init?(coder: NSCoder,
    	  weatherModel: WeatherModel,
    	  disasterModel: DisasterModel = DisasterModelImpl(),
    	  weatherHandler: WeatherHandler = .live,
    	  mainQueueScheduler: QueueScheduler = .main,
    	  globalQueueScheduler: QueueScheduler = .global) {
```